src/Core/Utilities/CustomRedisProcessingStrategy.cs

```diff
@@ -56,7 +58,7 @@ public override async Task<RateLimitCounter> ProcessRequestAsync(ClientRequestId
         if (_memoryCache.TryGetValue<TimeoutCounter>(_redisTimeoutCacheKey, out var timeoutCounter))
         {
             // We've exceeded threshold, backoff Redis and skip rate limiting for now
-            if (timeoutCounter.Count >= _distributedSettings.MaxRedisTimeoutsThreshold)
+            if (timeoutCounter?.Count >= _distributedSettings.MaxRedisTimeoutsThreshold)
```

Comment:
We shouldn't ever be caching a `null` value.

```suggestion
            Debug.Assert(timeoutCounter is not null);
            if (timeoutCounter.Count >= _distributedSettings.MaxRedisTimeoutsThreshold)
```